src/test/pages/EmailConfirmationModal.test.tsx

```diff
@@ -0,0 +1,93 @@
+import { screen, within } from "@testing-library/react";
+import produce from "immer";
+import { renderTestEnvironment } from "../utils";
+import userEvent from "@testing-library/user-event";
+import * as actions from "../../app/state/actions";
+import { rest } from "msw";
+import { API_PATH } from "../../app/services";
+
+const requestEmailVerificationSpy = jest.spyOn(
+  actions,
+  "requestEmailVerification"
+);
+
+describe("EmailConfirmationModal", () => {
+  it("If the user does not have a teacherPending: true flag, the modal will not appear", async () => {
+    renderTestEnvironment({
+      modifyUser: (user) =>
+        produce(user, (u) => {
+          u.role = "STUDENT";
+          u.emailVerificationStatus = "NOT_VERIFIED";
+        }),
+    });
+    const modal = screen.queryByTestId("active-modal");
```

Comment:
is there a risk that this test could pass because the modal hasn't rendered *yet* but eventually will? It might be necessary to check that rendering is actually complete before checking this.

Reply:
added waitFor which should resolve